src/components/landing/AnnouncementModal.tsx

```diff
@@ -0,0 +1,117 @@
+import { Announcement } from "@/models/Announcement";
+import AnnouncementBanner from "../shared/AnnouncementBanner";
+import AppIcon from "@/icons/AppIcon";
+import ButtonPrimary2 from "../shared/ButtonPrimary2";
+import ButtonPrimary3 from "../shared/ButtonPrimary3";
+import CrossThinIcon from "@/icons/CrossThinIcon";
+import { dateInRange, formatDate } from "@/utils/utils";
+import ModalLiveIndicator from "../shared/ModalLiveIndicator";
+import ModalPastIndicator from "../shared/ModalPastIndicator";
+import ModalUpcomingIndicator from "../shared/ModalUpcomingIndicator";
+import { useEffect, useState } from "react";
+
+interface AnnouncementModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  announcement: Announcement | null;
+}
+
+export default function AnnouncementModal({
+  isOpen,
+  onClose,
+  announcement,
+}: AnnouncementModalProps) {
+  if (!isOpen || !announcement) return null;
+
+  const [currentDate, setCurrentDate] = useState(new Date());
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setCurrentDate(new Date());
+    }, 500);
+
+    return () => clearInterval(intervalId);
+  }, []);
+
+  return (
+    <div className="fixed inset-0 bg-neutral-black bg-opacity-60 flex justify-center items-center z-50">
+      <div className="bg-neutral-white rounded-lg p-6 w-[500px] shadow-lg relative">
```

Comment:
```suggestion
      <div className="bg-neutral-white rounded-lg p-8 max-lg:w-full m-4 md:m-8 lg:max-w-[1128px]">
```
Sorry, this was not mentioned in the design but change it to the following so that it stretches and scales properly.